dlt/destinations/impl/synapse/synapse.py

```diff
@@ -0,0 +1,175 @@
+        if table is None:
+            return None
+        if table_name in self.schema.dlt_table_names():
+            # dlt tables should always be heap tables, regardless of the user
+            # configuration. Why? "For small lookup tables, less than 60 million rows,
+            # consider using HEAP or clustered index for faster query performance."
+            # https://learn.microsoft.com/en-us/azure/synapse-analytics/sql-data-warehouse/sql-data-warehouse-tables-index#heap-tables
+            table["table_index_type"] = "heap"
+        if table["table_index_type"] is None:
+            table["table_index_type"] = self.config.default_table_index_type
+        return table
+
+    def get_storage_table_index_type(self, table_name: str) -> TTableIndexType:
+        """Returns table index type of table in storage destination."""
+        with self.sql_client as sql_client:
+            schema_name = sql_client.fully_qualified_dataset_name(escape=False)
+            sql = dedent(f"""
+                SELECT
+                    CASE i.type_desc
+                        WHEN 'HEAP' THEN 'heap'
+                        WHEN 'CLUSTERED COLUMNSTORE' THEN 'clustered_columnstore_index'
+                    END AS table_index_type
+                FROM sys.indexes i
+                INNER JOIN sys.tables t ON t.object_id = i.object_id
+                INNER JOIN sys.schemas s ON s.schema_id = t.schema_id
+                WHERE s.name = '{schema_name}' AND t.name = '{table_name}'
+            """)
+            table_index_type = sql_client.execute_sql(sql)[0][0]
+            return cast(TTableIndexType, table_index_type)
+
+
+class SynapseStagingCopyJob(SqlStagingCopyJob):
+    @classmethod
+    def generate_sql(
+        cls,
+        table_chain: Sequence[TTableSchema],
+        sql_client: SqlClientBase[Any],
+        params: Optional[SqlJobParams] = None,
+    ) -> List[str]:
+        sql: List[str] = []
+        for table in table_chain:
+            with sql_client.with_staging_dataset(staging=True):
+                staging_table_name = sql_client.make_qualified_table_name(table["name"])
+            table_name = sql_client.make_qualified_table_name(table["name"])
+            # drop destination table
+            sql.append(f"DROP TABLE {table_name};")
+            # moving staging table to destination schema
+            sql.append(
+                f"ALTER SCHEMA {sql_client.fully_qualified_dataset_name()} TRANSFER"
+                f" {staging_table_name};"
+            )
+            # recreate staging table
+            # In some cases, when multiple instances of this CTAS query are
+            # executed concurrently, Synapse suspends the queries and hangs.
+            # This can be prevented by setting the env var LOAD__WORKERS = "1".
+            sql.append(
+                f"CREATE TABLE {staging_table_name}"
+                " WITH ( DISTRIBUTION = ROUND_ROBIN, HEAP )"  # distribution must be explicitly specified with CTAS
```

Comment:
could be configurable, but maybe let's jsut stick with this until someone requests otherwise. you can add a note in the docs about it

Reply:
I included a note in the docs.